packages/eslint-plugin/src/rules/no-confusing-void-expression.ts

```diff
@@ -376,5 +447,36 @@ export default createRule<Options, MessageId>({
       const type = getConstrainedTypeAtLocation(services, targetNode);
       return tsutils.isTypeFlagSet(type, ts.TypeFlags.VoidLike);
     }
+
+    function isVoidLikeType(typeNode?: TSESTree.TSTypeAnnotation): boolean {
+      const services = getParserServices(context);
+      return (
+        !!typeNode &&
+        !tsutils.isTypeFlagSet(
```

Comment:
Hi @developer-bandi 
It seems a typo. `isVoidLikeType` will return `false` when the given typeNode is void type.

```suggestion
        tsutils.isTypeFlagSet(
```

It can cause false positives such as the following case.

```ts
// ignoreVoidInVoid: true
 (): any => console.log('foo'); // no error
```

If there is no option, It's properly reported. [playground](https://typescript-eslint.io/play/#ts=5.4.2&fileType=.tsx&code=BQSgXABAhgdgnhAvAPggYwPYwM4YDYCmAdHhgObADkAZhhpSANxA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1rI6YDNYyZgHNaANw6UAJvQAexaCiG90URNGgdokcGAC%2BIPUA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)

Reply:
The code you presented was an obvious error, but it was naturally removed as the function was no longer used during the refactoring process below.

However, the exception case you presented was occurred because the return type did not expect a type other than void, and was resolved by adding logic to check whether the return type was void or unknown.